content/concepts/introduction/protocols/overview.md

```diff
@@ -1,206 +1,109 @@
+[Multistream-select](multistream.md), which routes the stream to the appropriate
+protocol handler based on the protocol ID.
+
+### Handler functions
+
+A handler function handles each libp2p protocol, responsible
+for defining the protocol's behavior once the stream has been established.
+The handler function is invoked when an incoming stream is received with a
+registered protocol ID.
+
+The handler function can also specify a match function,
+which allows for the acceptance of non-exact string matches for protocol IDs.
+
+A libp2p application will define a stream handler that takes over the
+stream after protocol negotiation. Everything is sent and received after the
+application protocol defines the negotiation phase.
+
+### Binary streams
+
+The medium over which a protocol operates is a bi-directional
+binary stream. This stream provides the following features:
+
+- **Bidirectional, reliable delivery of binary data**: Both peers can read and write
+  from the stream at any time, and data is read in the same order as it was written.
+  The stream can also be "half-closed", meaning it can be closed for writing while
+  still open for reading or closed for reading while still open for writing.
+- **Supports backpressure**: Eager writers cannot flood readers with data, as the
+  stream automatically regulates data flow to prevent overload.
+
+Behind the scenes, libp2p also ensures that the stream is securely encrypted and
+efficiently multiplexed, allowing multiple logical streams to be multiplexed over
+a single underlying connection. These details are transparent to the protocol handler,
+which reads and writes unencrypted binary data over the stream.
+
+## Life cycle of a stream
+
+The life cycle of a libp2p stream involves the following steps:
+
+- **Dialing out**: When a peer wants to initiate a new stream with another peer,
+  it sends the protocol ID of the protocol it wants to use over the connection.
+- **Protocol negotiation**: The listening peer on the other end checks the incoming
+  protocol ID against its list of registered protocol handlers. Suppose it does not
+  support the requested protocol. In that case, it sends "na" (not available) on the stream.
+  The dialing peer can try again with a different protocol or a fallback
+  version of the initially requested protocol. If the protocol is supported, the
+  listening peer echoes the protocol ID as a signal that future data sent over
+  the stream will use the agreed-upon protocol semantics.
+- **Stream establishment**: Once peers agree on the protocol ID, the stream is
+  established, and the designated invokes the handler function to take over the stream.
+  Everything sent and received over the stream from this point on is defined by the
+  application-level protocol.
+- **Stream closure**: When either peer finishes using the stream, it can be closed
+  by either side. If the stream is half-closed, the other side can continue to read
+  or write until it is closed.
```

Comment:
```suggestion
```
I would move this to the stream multiplexing document in a stream overview subsection
Is the protocols overview a good place for this?